atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/timestamp/InDbTimestampBoundStore.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright 2015 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.timestamp;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleErrorConstants;
+import com.palantir.common.base.Throwables;
+import com.palantir.exception.PalantirSqlException;
+import com.palantir.nexus.db.DBType;
+import com.palantir.nexus.db.pool.ConnectionManager;
+import com.palantir.nexus.db.pool.RetriableTransactions;
+import com.palantir.nexus.db.pool.RetriableTransactions.TransactionResult;
+import com.palantir.nexus.db.pool.RetriableWriteTransaction;
+import com.palantir.timestamp.MultipleRunningTimestampServiceError;
+import com.palantir.timestamp.TimestampBoundStore;
+
+// TODO: switch to using ptdatabase sql running, which more gracefully
+// supports multiple db types.
+public class InDbTimestampBoundStore implements TimestampBoundStore {
+    private static final Logger log = LoggerFactory.getLogger(InDbTimestampBoundStore.class);
+
+    private final ConnectionManager connManager;
+    private final TableReference timestampTable;
+    private final String tablePrefix;
+
+    @GuardedBy("this") // lazy init to avoid db connections in constructors
+    private DBType dbType;
+
+    @GuardedBy("this")
+    private Long currentLimit = null;
+
+    /**
+     * Use only if you have already initialized the timestamo table. This exists for legacy support.
+     */
+    public InDbTimestampBoundStore(ConnectionManager connManager, TableReference timestampTable) {
```

Comment:
This is reintroduced here in order to provide legacy support for our large internal product.  The replacement of this method with the constructor below was the breaking change.